.github/workflows/markdown-link-check.yml

```diff
@@ -1,7 +1,12 @@
 name: Check Markdown links
 on:
-  push:
   pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
```

Comment:
I still confused how we landed here. I thought I suggested that we want to check these *before* merging to main (so we don't introduce errors on main), and @esfoobar-usds suggested a very useful check that runs on pull requests? Did I miss some of the conversation? 

Thanks!

Reply:
@lucasmbrown-usds I believe the configuration as it is now will run whenever someone creates a pull request that is intended to merge into main, so it will catch that case.

This just means if we create a PR against someone else's branch, these won't run.

See https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#example-using-multiple-events-with-activity-types-or-configuration